### PR TITLE
GetLinuxNetworkInterfaces - avoid crash when /proc/net/route is inaccessible

### LINKED ISSUE #49946: GetLinuxNetworkInterfaces crash if /proc/net/route inaccessible
### Description

If the user running a dotnet application doesn't have permission to read `/proc/net/route`, and the application calls GetLinuxNetworkInterfaces, it crashes. One application where this is reproducible is Jellyfin, issue at jellyfin/jellyfin#5571

### Configuration

.NET 5.0.3
Linux (Debian x86_64 but should make no difference)

### Crash log

```
System.UnauthorizedAccessException: Access to the path '/proc/net/route' is denied.
 ---> System.IO.IOException: Permission denied
   --- End of inner exception stack trace ---
   at Interop.ThrowExceptionForIoErrno(ErrorInfo errorInfo, String path, Boolean isDirectory, Func`2 errorRewriter)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String path, OpenFlags flags, Int32 mode)
   at System.IO.FileStream.OpenHandle(FileMode mode, FileShare share, FileOptions options)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
   at System.IO.StreamReader.ValidateArgsAndOpenPath(String path, Encoding encoding, Int32 bufferSize)
   at System.IO.StreamReader..ctor(String path, Encoding encoding)
   at System.IO.File.InternalReadAllLines(String path, Encoding encoding)
   at System.IO.File.ReadAllLines(String path)
   at System.Net.NetworkInformation.LinuxNetworkInterface.LinuxNetworkInterfaceSystemProperties..ctor()
   at System.Net.NetworkInformation.LinuxNetworkInterface.GetLinuxNetworkInterfaces()
```

### Other information

[GetLinuxNetworkInterfaces](https://github.com/dotnet/runtime/blob/79ae74f5ca5c8a6fe3a48935e85bd7374959c570/src/libraries/System.Net.NetworkInformation/src/System/Net/NetworkInformation/LinuxNetworkInterface.cs#L23) seems to check if the file exists before trying to read it but it doesn't check if it actually is allowed to be read. A workaround is to remove permissions from the directory /proc/net so .NET can't validate the file's existence and it just skips the file. A simple solution would be to just add another if statement to check if the file can be read. Or catch the UnauthorizedAccessException.

### PR DESCRIPTION
**Pull request by karelz**
Ignore UnauthorizedAccessException when reading route files.

Fixes #49946

### CODE REVIEW

File: src/libraries/System.Net.NetworkInformation/src/System/Net/NetworkInformation/LinuxNetworkInterface.cs [hunk 1]
```diff
@@ -31,12 +31,24 @@ internal LinuxNetworkInterfaceSystemProperties()
...
                 if (File.Exists(NetworkFiles.Ipv6RouteFile))
                 {
-                    IPv6Routes = File.ReadAllLines(NetworkFiles.Ipv6RouteFile);
+                    try
+                    {
+                        IPv6Routes = File.ReadAllLines(NetworkFiles.Ipv6RouteFile);
+                    }
+                    catch (UnauthorizedAccessException)
+                    {
+                    }
```
Comment (kasperk81) [inline src/libraries/System.Net.NetworkInformation/src/System/Net/NetworkInformation/LinuxNetworkInterface.cs:51]: line 70 too?
```c#
catch (Exception e) when (e is FileNotFoundException || e is UnauthorizedAccessException)
```

Comment (karelz) [reply src/libraries/System.Net.NetworkInformation/src/System/Net/NetworkInformation/LinuxNetworkInterface.cs:51]: I don't see anything related on line 70 ...
We already test for existence of the file, no need to change that into an exception IMO.

Comment (kasperk81) [reply src/libraries/System.Net.NetworkInformation/src/System/Net/NetworkInformation/LinuxNetworkInterface.cs:51]: ops, it was line 60 for EtcResolvConfFile.

Comment (karelz) [reply src/libraries/System.Net.NetworkInformation/src/System/Net/NetworkInformation/LinuxNetworkInterface.cs:51]: Yeah, I was considering that one earlier and then I forgot about it. Given that we can already live without the file, we can probably extend it also to inaccessible file ...

Comment (karelz) [reply src/libraries/System.Net.NetworkInformation/src/System/Net/NetworkInformation/LinuxNetworkInterface.cs:51]: Thanks for pointing it out @kasperk81! The PR is now updated.

**Comment by karelz** [comment]
@wfurt can you please review it?

**Review by wfurt — APPROVED** [review]
LGTM

**Pull request merged**